spark-common/src/main/js/src/spark-common-templates.js

```diff
@@ -21,25 +43,32 @@ const appBootstrapContaineriFrame = function({id, src, createOptions, className}
  * @param createOptions
  * @param className
  */
-const appBootstrapContainerDialog2WithiFrame = function({id, title, src, createOptions, className}) {
+const appBootstrapContainerDialog2WithiFrame = function(options) {
+    const { id, title, src, className } = escapeHtmlMany(options, ['id', 'title', 'src', 'className']);
+    const createOptions = options.createOptions;
+    const width = escapeHtml(createOptions.width);
+    const height = escapeHtml(createOptions.height);
+    const showSubmitButton = !!createOptions.showSubmitButton;
+    const submitLabel = escapeHtml(createOptions.label.submit);
+    const closeLabel = escapeHtml(createOptions.label.close);
     return `
-        <section role="dialog" id="${id}" class="${className} aui-layer aui-dialog2" style="width:${createOptions.width};"
+        <section role="dialog" id="${id}" class="${className} aui-layer aui-dialog2" style="width:${width};"
                 aria-hidden="true">
             <header class="aui-dialog2-header">
                 <h2 class="aui-dialog2-header-main">${title}</h2>
             </header>
             <div class="aui-dialog2-content spark-app-content"
-                    style="padding: 0; width:${createOptions.width}; height: ${createOptions.height}; overflow: hidden;">
-                ${appBootstrapContaineriFrame({id, src, createOptions, className})}
+                    style="padding: 0; width:${width}; height: ${height}; overflow: hidden;">
+                ${appBootstrapContaineriFrame({ id, src, createOptions, className })}
```

Comment:
Looks like you're invoking this function with the escaped values, yet the function internally escapes again.
I think we should rather pass the unescaped values here.